flake8-builtins is not happy with some of our names

When running tests:

```
=================================== FAILURES ===================================
_________________________________ FLAKE8-check _________________________________
/Users/sobolev/Documents/github/mimesis/mimesis/providers/business.py:45:5: A003 "copyright" is a python builtin, consider renaming the class attribute

_________________________________ FLAKE8-check _________________________________
/Users/sobolev/Documents/github/mimesis/mimesis/providers/cryptographic.py:37:5: A003 "hash" is a python builtin, consider renaming the class attribute
/Users/sobolev/Documents/github/mimesis/mimesis/providers/cryptographic.py:50:5: A003 "bytes" is a python builtin, consider renaming the class attribute

============== 2 failed, 4067 passed, 1 skipped in 37.43 seconds ===============
```

So, I will just ignore these issues. I do not what to change something or break the API.

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -23,7 +23,7 @@ install:
 
   # Upgrade to the latest version of pip to avoid it displaying warnings
   # about it being out of date.
-  - "pip install --disable-pip-version-check --user --upgrade pip"
+  - "python -m pip install --disable-pip-version-check --user --upgrade pip"
 
   # Setting up enviroment
   - "pip install pipenv"

--- a/mimesis/helpers.py
+++ b/mimesis/helpers.py
@@ -10,13 +10,13 @@ get a random item of the enum object.
 """
 
 import os
-import random
+import random as random_module
 from typing import Any, List, Optional, Sequence
 
 __all__ = ['Random', 'get_random_item']
 
 
-class Random(random.Random):
+class Random(random_module.Random):
     """Custom class for the possibility of extending.
 
     The class is a subclass of the class ``Random()`` from the module ``random``
@@ -119,4 +119,9 @@ def get_random_item(enum: Any, rnd: Optional[Random] = None) -> Any:
     """
     if rnd and isinstance(rnd, Random):
         return rnd.choice(list(enum))
-    return random.choice(list(enum))
+    return random_module.choice(list(enum))
+
+
+# Compat
+# See: https://github.com/lk-geimfari/mimesis/issues/469
+random = Random()

--- a/mimesis/providers/base.py
+++ b/mimesis/providers/base.py
@@ -17,6 +17,7 @@ class BaseProvider(object):
         """Initialize attributes.
 
         :param seed: Seed for random.
+            When set to `None` the current system time is used.
         """
         self.seed = seed
         self.random = random
@@ -24,7 +25,7 @@ class BaseProvider(object):
         if seed is not None:
             self.reseed(seed)
 
-    def reseed(self, seed: Seed) -> None:
+    def reseed(self, seed: Optional[Seed]) -> None:
         """Reseed the internal random generator.
 
         In case we use the default seed, we need to create a per instance
@@ -32,6 +33,7 @@ class BaseProvider(object):
         will always return the same values.
 
         :param seed: Seed for random.
+            When set to `None` the current system time is used.
         """
         if self.random is random:
             self.random = Random()

--- a/mimesis/providers/base.py
+++ b/mimesis/providers/base.py
@@ -3,7 +3,7 @@
 from typing import Any, Optional
 
 from mimesis.exceptions import NonEnumerableError
-from mimesis.helpers import Random, get_random_item
+from mimesis.helpers import Random, get_random_item, random
 from mimesis.typing import Seed
 from mimesis.utils import setup_locale
 
@@ -19,10 +19,25 @@ class BaseProvider(object):
         :param seed: Seed for random.
         """
         self.seed = seed
-        self.random = Random()
+        self.random = random
 
         if seed is not None:
-            self.random.seed(self.seed)
+            self.reseed(seed)
+
+    def reseed(self, seed: Seed) -> None:
+        """Reseed the internal random generator.
+
+        In case we use the default seed, we need to create a per instance
+        random generator, in this case two providers with the same seed
+        will always return the same values.
+
+        :param seed: Seed for random.
+        """
+        if self.random is random:
+            self.random = Random()
+
+        self.seed = seed
+        self.random.seed(self.seed)
 
     def _validate_enum(self, item: Any, enum: Any) -> Any:
         """Validate enum parameter of method in subclasses of BaseProvider.

--- a/mimesis/providers/business.py
+++ b/mimesis/providers/business.py
@@ -42,7 +42,7 @@ class Business(BaseDataProvider):
             self._data['company']['type'].get('abbr' if abbr else 'title'),
         )
 
-    def copyright(self) -> str:  # noqa: A002
+    def copyright(self) -> str:  # noqa: A003
         """Generate a random copyright.
 
         :return: Copyright of company.

--- a/mimesis/providers/cryptographic.py
+++ b/mimesis/providers/cryptographic.py
@@ -34,7 +34,7 @@ class Cryptographic(BaseDataProvider):
         bits = self.random.getrandbits(128)
         return str(uuid.UUID(int=bits, version=version))
 
-    def hash(self, algorithm: Optional[Algorithm] = None) -> str:  # noqa: A002
+    def hash(self, algorithm: Optional[Algorithm] = None) -> str:  # noqa: A003
         """Generate random hash.
 
         :param algorithm: Enum object ``Algorithm``.
@@ -47,7 +47,7 @@ class Cryptographic(BaseDataProvider):
             fn = getattr(hashlib, key)
             return fn(self.uuid().encode()).hexdigest()
 
-    def bytes(self, entropy: int = 32) -> Bytes:  # noqa: A002
+    def bytes(self, entropy: int = 32) -> Bytes:  # noqa: A003
         """Generate byte string containing *entropy* bytes.
 
         The string has *entropy* random bytes, each byte

--- a/tests/test_helpers.py
+++ b/tests/test_helpers.py
@@ -1,12 +1,13 @@
 import pytest
 
 from mimesis.enums import Gender
-from mimesis.helpers import Random, get_random_item
+from mimesis.helpers import get_random_item
+from mimesis.helpers import random as _random
 
 
 @pytest.fixture
 def random():
-    return Random()
+    return _random
 
 
 def test_randints(random):

